pkgs/development/python-modules/babeldoc/default.nix

```diff
@@ -0,0 +1,121 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  fetchpatch2,
+
+  # build dependencies
+  hatchling,
+
+  # dependencies
+  bitstring,
+  configargparse,
+  httpx,
+  huggingface-hub,
+  numpy,
+  onnx,
+  onnxruntime,
+  openai,
+  orjson,
+  charset-normalizer,
+  cryptography,
+  peewee,
+  psutil,
+  pymupdf,
+  rich,
+  toml,
+  tqdm,
+  xsdata,
+  msgpack,
+  pydantic,
+  tenacity,
+  scikit-image,
+  freetype-py,
+  tiktoken,
+  levenshtein,
+  opencv-python-headless,
+  rapidocr-onnxruntime,
+  pyzstd,
+  hyperscan,
+  rtree,
+  chardet,
+  scipy,
+  uharfbuzz,
+  scikit-learn,
+}:
+
+buildPythonPackage rec {
+  pname = "babeldoc";
+  version = "0.5.21";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "funstory-ai";
+    repo = "BabelDOC";
+    tag = "v${version}";
+    hash = "sha256-vBlrsJN/VkwmpBzZAqyZAxz5W2vgWx/jbUzvyzeKtUc=";
+  };
+
+  patches = [
+    (fetchpatch2 {
+      name = "rename-python-levenshtein-to-levenshtein";
+      url = "https://github.com/funstory-ai/BabelDOC/pull/542.patch";
+      hash = "sha256-tPgXZdKNOA1fJeFEE3PVVwWZZ+jLkW1bfyymhFlVXE8=";
+    })
+  ];
+
+  build-system = [ hatchling ];
+
+  dependencies = [
+    bitstring
+    configargparse
+    httpx
+    huggingface-hub
+    numpy
+    onnx
+    onnxruntime
+    openai
+    orjson
+    charset-normalizer
+    cryptography
+    peewee
+    psutil
+    pymupdf
+    rich
+    toml
+    tqdm
+    xsdata
+    msgpack
+    pydantic
+    tenacity
+    scikit-image
+    freetype-py
+    tiktoken
+    levenshtein
+    opencv-python-headless
+    rapidocr-onnxruntime
+    pyzstd
+    hyperscan
+    rtree
+    chardet
+    scipy
+    uharfbuzz
+    scikit-learn
+  ]
+  ++ httpx.optional-dependencies.socks
+  ++ (with xsdata.optional-dependencies; cli ++ lxml ++ soap);
+
+  pythonImportsCheck = [ "babeldoc" ];
+
+  meta = {
+    description = "Yet Another Document Translator";
+    homepage = "https://github.com/funstory-ai/BabelDOC";
+    platforms = [
+      "x86_64-linux"
+      "x86_64-darwin"
```

Comment:
Why so restrictive? I don't see any non-Python code in the repo, so any restrictive `platforms` should be specified by dependencies.